core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3443,6 +3443,9 @@ class ViewMapping<
     const memory_space& mem_space =
         Impl::get_property<Impl::MemorySpaceTag>(arg_prop);
 
+    static_assert(
+        SpaceAccessibility<execution_space, memory_space>::accessible);
```

Comment:
By the way why did you place this static assertion here rather than right after `execution_space` and `memory_space` definitions?

Reply:
I didn't want to interrupt the block of aliases and it felt like a good place since we are declaring related variables close to it.

Reply:
Move it right after their definition (L3421)